Issue/3844 Improves the appearance of the posts / pages list footer animation

frosty opened the pull request:
Refs #3844.

I think ultimately the original issue is probably due to the estimated height for cells being different to the final height (just a guess), however these few tweaks improve the animation a lot. I've simply given the cells clear backgrounds and moved the footer view to the back. 

Before:

![footer-bug](https://cloud.githubusercontent.com/assets/4780/16683492/184eb640-44f8-11e6-8990-0d26f01e9012.gif)

After:

![footer-bug-2](https://cloud.githubusercontent.com/assets/4780/16683496/1f9ee65e-44f8-11e6-9828-87202dc9b797.gif)

### To test

I was able to replicate this simply by viewing the Posts list on a site with a single post, and the effect is easier to see with slow animations enabled. I guess also check you can't see any negative effects of making the post cell backgrounds clear.

Do you think this is an acceptable 'fix' for the original issue, or should we also look at preventing the animation in the first place?

Needs review: @aerych

aerych commented:
> I think ultimately the original issue is probably due to the estimated height for cells being different to the final height (just a guess),

Its exactly this, and the estimated height can also cause a similarly funky animation of cells "cascading" when the vc is first pushed (tho I don't always see it).

Rather than fussing with the stack order, maybe we just not return an estimated height for the first _n_ cells.  Something like this seems to resolve both issues yet still retains the improved scrolling performance from using estimated heights:
```
    func tableView(tableView: UITableView, estimatedHeightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        let post = postAtIndexPath(indexPath)

        if cellIdentifierForPost(post) == self.dynamicType.postCardRestoreCellIdentifier {
            return self.dynamicType.postCardRestoreCellRowHeight
        }

        if indexPath.row < 5 {
            return self.tableView(tableView, heightForRowAtIndexPath: indexPath)
        }
        return self.dynamicType.postCardEstimatedRowHeight
    }
```
Thoughts?

frosty commented:
@aerych This seems to work well! The question is, how do we choose what '_n_' should be? It looks as though we could fit 7 post cards on a vertical iPad Pro 12" screen (with one line of text each). Should we go with that? Should we calculate it based on the screen's height and the height of a cell that we populate with a single line of text?

aerych commented:
> Should we calculate it based on the screen's height and the height of a cell that we populate with a single line of text

That sounds good to me. :)

frosty commented:
@aerych Sorry for the delay in getting back to this! I've reverted my previous changes, and switched to the method we discussed. It seems to work fine! Would you mind taking another look?

aerych commented:
Looks good :)
:shipit: sir!
